httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt

```diff
@@ -64,7 +68,70 @@ object TbdexHttpClient {
         val jsonNode = jsonMapper.readTree(responseString)
         return jsonNode.get("data").elements()
           .asSequence()
-          .map { Offering.parse(it.toString()) }
+          .map {
+            try {
+              Offering.parse(it.toString())
+            } catch (e: Exception) {
+              throw TbdexResponseException(
+                "response status: ${response.code}",
+                errors = listOf(
+                  ErrorDetail(
+                    detail = "Failed to parse offering ${e.message}"
+                  )
+                )
+              )
+            }
+          }
+          .toList()
+      }
+
+      else -> throw buildResponseException(response)
+    }
+  }
+
+  /**
+   * Fetches balances from a PFI.
+   *
+   * @param pfiDid The decentralized identifier of the PFI.
+   * @param requesterDid The decentralized identifier of the entity requesting the balances.
+   * @return A list of [Balance] matching the request.
+   * @throws TbdexResponseException for request or response errors.
+   */
+  @Suppress("SwallowedException")
+  fun getBalances(pfiDid: String, requesterDid: BearerDid): List<Balance> {
+    val pfiServiceEndpoint = getPfiServiceEndpoint(pfiDid)
+    val baseUrl = "$pfiServiceEndpoint/balances/"
+    val requestToken = RequestToken.generate(requesterDid, pfiDid)
+
+    val request = Request.Builder()
+      .url(baseUrl)
+      .addHeader(CONTENT_TYPE, JSON_HEADER)
+      .addHeader(AUTHORIZATION, "Bearer $requestToken")
+      .get()
+      .build()
+
+    val response: Response = client.newCall(request).execute()
```

Comment:
ah does this throw an exception if the request fails? e.g. timeout, `EHANGUP` etc.

Reply:
correct. from the docs:

```
Throws:
IOException - if the request could not be executed due to cancellation, 
a connectivity problem or timeout. Because networks can fail during an exchange,
it is possible that the remote server accepted the request before the failure.
IllegalStateException - when the call has already been executed.
```